octopus_deploy/datadog_checks/octopus_deploy/check.py

```diff
@@ -341,6 +342,29 @@ def _collect_server_nodes_metrics(self):
             self.gauge("server_node.in_maintenance_mode", maintenance_mode, tags=self._base_tags + server_tags)
             self.gauge("server_node.max_concurrent_tasks", max_tasks, tags=self._base_tags + server_tags)
 
+    def _collect_machine_metrics(self, space_id):
+        self.log.debug("Collecting server node metrics.")
+        url = f"api/{space_id}/machines"
+        response_json = self._process_paginated_endpoint(url)
+        machines = response_json.get('Items', [])
+
+        for machine in machines:
+            machine_id = machine.get("Id")
+            machine_name = machine.get("Name")
+            machine_slug = machine.get("Slug")
+            roles = machine.get("Roles", [])
+            health_status = machine.get("HealthStatus", None)
+            is_healthy = health_status.startswith("Healthy")
+            machine_tags = [
+                f"machine_id:{machine_id}",
+                f"machine_name:{machine_name}",
+                f"machine_slug:{machine_slug}",
+                f"health_status:{health_status}",
+            ]
+            machine_tags += roles
+            self.gauge("machine.count", 1, tags=self._base_tags + machine_tags)
+            self.gauge("machine.is_healthy", is_healthy, tags=self._base_tags + machine_tags)
```

Comment:
What is the purpose of also reporting it as a metric if we already report it as a tag?

Reply:
It's easier to monitor on, and the tag gives more specific/detailed information